backend/FwLite/LcmCrdt/Changes/Entries/SetComplexFormComponentChange.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Text.Json.Serialization;
+using LcmCrdt.Objects;
+using SIL.Harmony.Changes;
+using SIL.Harmony.Entities;
+
+namespace LcmCrdt.Changes.Entries;
+
+public class SetComplexFormComponentChange : EditChange<CrdtComplexFormComponent>, ISelfNamedType<SetComplexFormComponentChange>
+{
+    [JsonConstructor]
+    protected SetComplexFormComponentChange(Guid entityId, Guid? complexFormEntryId, Guid? componentEntryId, Guid? componentSenseId) : base(entityId)
+    {
+        ComplexFormEntryId = complexFormEntryId;
+        ComponentEntryId = componentEntryId;
+        ComponentSenseId = componentSenseId;
+    }
+
+    public static SetComplexFormComponentChange NewComplexForm(Guid id, Guid complexFormEntryId) => new(id, complexFormEntryId, null, null);
+    public static SetComplexFormComponentChange NewComponent(Guid id, Guid componentEntryId) => new(id, null, componentEntryId, null);
+    public static SetComplexFormComponentChange NewComponentSense(Guid id, Guid componentEntryId, Guid? componentSenseId) => new(id, null, componentEntryId, componentSenseId);
+    public Guid? ComplexFormEntryId { get; }
+    public Guid? ComponentEntryId { get; }
+    public Guid? ComponentSenseId { get; }
+    public override async ValueTask ApplyChange(CrdtComplexFormComponent entity, ChangeContext context)
+    {
+        if (ComplexFormEntryId.HasValue)
+        {
+            entity.ComplexFormEntryId = ComplexFormEntryId.Value;
```

Comment:
Don't we need to update headwords here? We could presumably pull them out of the context.
Or are you just ignoring them, because we don't know how to update them when the headword of a referenced entry changes?

Reply:
good point, I can do that actually.